src/auto-evo/RunResults.cs

```diff
@@ -793,6 +797,24 @@ void OutputPopulationForPatch(Species species, Patch patch, long population)
                     builder.Append('\n');
             }
 
+            highlightSpecies ??= new HashSet<Species>();
+            if (highlightSpecies.Contains(null))
+                throw new ArgumentNullException("Highlight species entry must not be null");
+
+            // Print highlight species results
+            foreach (Species species in highlightSpecies.OrderBy(s => s.FormattedName))
+            {
+                if (results.ContainsKey(species))
+                    AddSpeciesResultString(results[species]);
```

Comment:
The dictionary method `TryGetValue(key, out var value)` is precisely designed for these cases, allowing you to define `dict[key]` in `var value`, which you can then use (even outside the `if` statement, don't ask me about this magic).

This is both easier to read and more efficient computationally.

Reply:
Thanks, I will keep that in mind. I guess I have a lot to learn, especially when it comes to performance sensitive operations like in a game.

Reply:
That's normal, we all learn a lot thanks to Thrive. Actually, I've especially just got to learn a lot about this method with my latest PRs. 😉